test/integration/network_reserved_ips_test.go

```diff
@@ -0,0 +1,284 @@
+package integration
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	. "github.com/linode/linodego"
+)
+
+// TestReservedIPAddresses_InsufficientPermissions tests the behavior when a user account
+// doesn't have the can_reserve_ip flag enabled
+func TestReservedIPAddresses_InsufficientPermissions(t *testing.T) {
+	original := os.Getenv("LINODE_TOKEN")
+	dummyToken := "badtoken"
+	os.Setenv("LINODE_TOKEN", dummyToken)
+
+	client, teardown := createTestClient(t, "fixtures/TestReservedIpAddresses_InsufficientPermissions")
+	defer teardown()
+	defer os.Setenv("LINODE_TOKEN", original)
+
+	filter := ""
+	i, getReservedIpsError := client.GetReservedIPs(context.Background(), NewListOptions(0, filter))
+	if getReservedIpsError != nil {
+		t.Logf("Error listing ipaddresses, expected struct, got error %v", getReservedIpsError)
+	}
+
+	if len(i) == 0 {
+		t.Logf("Expected a list of ipaddresses, but got none %v", i)
+	}
+
+	// Attempt to reserve an IP address
```

Comment:
You expect this to fail, right? You should probably fail the test in the case you get a positive result.

Reply:
Yes I have now used errorf() insetad of logf to fail the test in case the expected error does not occur.